### ISSUE TITLE
jsonpath for JSON Extra-files in yaml config

### ISSUE THREAD

**Issue by fire-cracker** [google-github-actions/release-please-action#797]
### TL;DR

Enable the usage of JSONPath to annotate properties in YAML configuration when specifying extra-files of type JSON for release-please, similar to the existing functionality in release-please-config.json.

### Detailed design

Proposed Solution;

      - name: Create Release
        uses: google-github-actions/release-please-action@v3
        with:
          release-type: node
          command: release-pr
          extra-files:
            - path: public/manifest.json
              type: json
              jsonpath: $.version

### Additional information

Annotations such as the one provided in the docs are not possible when dealing with JSON files. The request is made as I would like release-please to be able to update my Manifest file version.

**Comment by chingor13** [comment]
For complex configurations (including extra-files), please [configure a release-please manifest configuration](https://github.com/googleapis/release-please/blob/main/docs/manifest-releaser.md#configfile). 

The configuration is getting very complex and it's not maintainable to map every complex configuration option into GitHub action yaml input counterparts. We don't plan on proactively trying to map this inputs moving forward.
